cli/account_tls_command.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2022 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cli
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/choria-io/fisk"
+
+	"golang.org/x/crypto/ocsp"
+)
+
+type ActTLSCmd struct {
+	expireWarnDuration time.Duration
+	wantOCSP           bool
+	wantPEM            bool
+
+	// values after here derived inside show_tls
+	now          time.Time
+	warnIfBefore time.Time
+}
+
+func configureAccountTLSCommand(srv *fisk.CmdClause) {
+	c := &ActTLSCmd{}
+
+	tls := srv.Command("tls", "Report TLS chain for connected server").Action(c.show_tls)
+	tls.Flag("expire-warn", "Warn about certs expiring this soon (1w; 0 to disable)").Default("1w").PlaceHolder("DURATION").DurationVar(&c.expireWarnDuration)
+	tls.Flag("ocsp", "Report OCSP information (if any)").UnNegatableBoolVar(&c.wantOCSP)
+	tls.Flag("pem", "Show PEM Certificate blocks").UnNegatableBoolVar(&c.wantPEM)
+
+	// TODO: consider NAGIOS-compatible option (output format, exit statuses)
+}
+
+func (c *ActTLSCmd) show_tls(_ *fisk.ParseContext) error {
+	c.now = time.Now()
+	if c.expireWarnDuration > 0 {
+		c.warnIfBefore = c.now.Add(c.expireWarnDuration)
+	}
+
+	nc, _, err := prepareHelper("", natsOpts()...)
+	if err != nil {
+		return err
+	}
+
+	t, err := nc.TLSConnectionState()
+	if err != nil {
+		return err
+	}
+
+	var showingOCSP bool
+	if c.wantOCSP {
+		if t.OCSPResponse != nil && len(t.OCSPResponse) > 0 {
+			showingOCSP = true
+		} else {
+			fmt.Printf("# No OCSP Response found in TLS connection\n\n")
+		}
+	}
+
+	fmt.Printf("# TLS Verified Chains count: %d\n", len(t.VerifiedChains))
+	if len(t.VerifiedChains) < 1 {
+		return fmt.Errorf("no verified chains found in TLS")
+	}
+
+	err = nil
+	for i := range t.VerifiedChains {
+		fmt.Printf("\n# chain: %d\n", i+1)
+		if chainErr := c.show_one_tls_chain(t.VerifiedChains[i], i+1); chainErr != nil && err == nil {
+			err = chainErr
+		}
+		if showingOCSP {
+			if ocspErr := c.show_one_ocsp(t.VerifiedChains[i], i+1, t); ocspErr != nil && err == nil {
+				err = ocspErr
+			}
+		}
+	}
+	return err
+}
+
+func (c *ActTLSCmd) show_one_tls_chain(chain []*x509.Certificate, chain_number int) error {
```

Comment:
goCode not perl_code :)

Reply:
Dropped the Pythonic snake-case, which I thought was matching existing style and had me raise half an eye-brow, but I was thinking more about the code at the time and didn't check and I'd mis-read, so I wasn't matching style.